lms/services/group_set.py

```diff
@@ -0,0 +1,51 @@
+from typing import TypedDict
+
+from lms.models.group_set import LMSGroupSet
+from lms.services.upsert import bulk_upsert
+
+
+class GroupSetDict(TypedDict):
+    """
+    Group sets are a collection of student groups.
+
+    We store them in Course.extra
+    """
+
+    id: str
+    name: str
+
+
+class GroupSetService:
+    def __init__(self, db):
+        self._db = db
+
+    def store_group_sets(self, course, group_sets: list[dict]):
+        """
+        Store this course's available group sets.
+
+        We keep record of these for bookkeeping and as the basics to
+        dealt with groups while doing course copy.
+        """
+        # Different LMS might return additional fields but we only interested in the ID and the name.
+        # We explicitly cast ID to string to homogenise the data in all LMS's.
+        group_sets = [{"id": str(g["id"]), "name": g["name"]} for g in group_sets]
+        course.extra["group_sets"] = group_sets
```

Comment:
Do we need to continue saving group sets in course.extra? Maybe if we stop saving it, we can know which courses were created before this change and simplifies backfilling the lms_group_sets table.

Reply:
I think we need to as we are doing reads too, once the backfill is done we can switch the reads to the new table and then stop writing to extra :+1:

Reply:
Makes sense